src/common/requests.ts

```diff
@@ -133,6 +133,15 @@ function makeRequestsDeclaration(
   );
 }
 
+type HttpMethods =
```

Comment:
Would you be able to infer this type based on the array created below? Would reduce the repetitive declarations. Something like `type HttpMethods = typeof methods[number]` and the `methods` array should be defined like `const methods = ['get', 'post', ...] as const` <- the `as const` is key for the type to be inferred properly